src/spyglass/utils/nwb_helper_fn.py

```diff
@@ -22,81 +22,104 @@
 invalid_electrode_index = 99999999
 
 
-def get_nwb_file(nwb_file_path):
+def _open_nwb_file(nwb_file_path, source="local"):
+    """Open an NWB file, add to cache, return contents. Does not close file."""
+    if source == "local":
+        io = pynwb.NWBHDF5IO(path=nwb_file_path, mode="r", load_namespaces=True)
+        nwbfile = io.read()
+    elif source == "dandi":
+        from ..common.common_dandi import DandiPath
+
+        io, nwbfile = DandiPath().fetch_file_from_dandi(
+            nwb_file_path=nwb_file_path
+        )
+    else:
+        raise ValueError(f"Invalid open_nwb source: {source}")
+    __open_nwb_files[nwb_file_path] = (io, nwbfile)
+    return nwbfile
+
+
+def get_nwb_file(nwb_file_path, query_expression=None):
     """Return an NWBFile object with the given file path in read mode.
 
     If the file is not found locally, this will check if it has been shared
-    with kachery and if so, download it and open it.
-
+    with kachery/dandi and if so, download it and open it. If not, and the
+    query_expression has a `_make_file` method, it will call that method to
+    recompute the file.
 
     Parameters
     ----------
     nwb_file_path : str
         Path to the NWB file or NWB file name. If it does not start with a "/",
         get path with Nwbfile.get_abs_path
+    query_expression : dj.QueryExpression, optional
+        The restricted table associated with the NWB file, used for recompute.
 
     Returns
     -------
     nwbfile : pynwb.NWBFile
         NWB file object for the given path opened in read mode.
+
+    Raises
+    ------
+    FileNotFoundError
+        If the NWB file is not found locally or in kachery/Dandi, and cannot be
+        recomputed.
     """
-    if not nwb_file_path.startswith("/"):
-        from ..common import Nwbfile
+    if not str(nwb_file_path).startswith("/"):
```

Comment:
Assumes a linux based system?

Reply:
I was operating on the assumption that we only support unix file systems

https://github.com/LorenFrankLab/spyglass/blob/46d59d654e1e2ec5377c37481c0809b7a915b409/README.md?plain=1#L95
 
 Easy enough to swap out for a Path.is_abs check though

Reply:
I just prefer to not exclude windows based systems with paths if we don't have to. There might be a future in which we do support it.